### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by kjbranch**

### CODE REVIEW

File: tests/asserts/keywords.py [hunk 1]
```diff
@@ -372,8 +475,97 @@ def process_strats(src, paramData):
...
+                            # Validate Requires Nodes
+                            requiresErrors = search_for_valid_keyvalue(
+                                [
+                                    "fromNode",
+                                    "fromNodes.",
+                                    "inRoomPath.",
+                                    "resetRoom.nodes.",
+                                    "nodesToAvoid.",
+                                    "itemNotCollectedAtNode",
+                                    "entranceNodes."
```
Comment (osse101) [inline tests/asserts/keywords.py:552]: entranceNodes is part of the room description
`rooms: roomEnvironments: entranceNodes: []`

The node value would be a door (way of entering) node for the room.

File: tests/asserts/keywords.py [hunk 2]
```diff
@@ -398,6 +591,17 @@ def process_strats(src, paramData):
                                                             msg = f"🔴ERROR: Destination node not defined:{fromNodeRef}"
                                                             messages["reds"].append(msg)
                                                             messages["counts"]["reds"] += 1
+                                                        if str(fromNode) in roomData["links"]["from"]:
```
Comment (osse101) [inline tests/asserts/keywords.py:594]: Does `fromNode` update to become the previous "destinationNode"?

File: tests/asserts/keywords.py (hunk 1 above)
Comment (miketrethewey) [reply tests/asserts/keywords.py]: I've moved `entranceNodes` to it's own check and have it check against nodes that have `nodeType` `door` or `entrance`.
It'll be included in an upcoming commit.

File: tests/asserts/keywords.py (hunk 2 above)
Comment (miketrethewey) [reply tests/asserts/keywords.py:702]: I changed this to leverage `search_for_path()` which will try to find a multi-node pathway.
It'll be included in an upcoming commit.

**Review by osse101 — DISMISSED** [review]

**Pull request merged**